pkg/agent/proxy/proxier_test.go

```diff
@@ -481,20 +481,28 @@ func newFakeProxier(routeClient route.Interface, ofClient openflow.Client, nodeP
 	if o.serviceProxyNameSet {
 		serviceProxyName = testServiceProxyName
 	}
-	fakeClient := fake.NewSimpleClientset()
-	fakeNodeIPChecker := nodeipmock.NewFakeNodeIPChecker()
-	informerFactory := informers.NewSharedInformerFactory(fakeClient, 0)
+	topologyAwareHintsEnabled := o.endpointSliceEnabled && features.DefaultFeatureGate.Enabled(features.TopologyAwareHints)
+	serviceTrafficDistributionEnabled := o.endpointSliceEnabled && features.DefaultFeatureGate.Enabled(features.ServiceTrafficDistribution)
+
+	// TODO: The label selector nonHeadlessServiceSelector was added to pass the Kubernetes e2e test
+	//  'Services should implement service.kubernetes.io/headless'. You can find the test case at:
+	//  https://github.com/kubernetes/kubernetes/blob/027ac5a426a261ba6b66a40e79e123e75e9baf5b/test/e2e/network/service.go#L2281
+	//  However, in AntreaProxy, headless Services are skipped by checking the ClusterIP.
+	nonHeadlessServiceSelector, _ := labels.NewRequirement(corev1.IsHeadlessService, selection.DoesNotExist, nil)
+	var serviceProxyNameSelector *labels.Requirement
+	if serviceProxyName == "" {
+		serviceProxyNameSelector, _ = labels.NewRequirement(labelServiceProxyName, selection.DoesNotExist, nil)
+	} else {
+		serviceProxyNameSelector, _ = labels.NewRequirement(labelServiceProxyName, selection.DoubleEquals, []string{serviceProxyName})
+	}
+	serviceLabelSelector := labels.NewSelector()
+	serviceLabelSelector = serviceLabelSelector.Add(*serviceProxyNameSelector, *nonHeadlessServiceSelector)
```

Comment:
it seems wrong to have to repeat that in test code to initialize the proxier

I also wonder if we are losing a lot of unit test coverage because some logic was moved to the new `ProxyServer` and the test code is not using that object but the (now) lower-level `proxier`?

Reply:
Will add new tests to cover higher-level `ProxyServer`.